tests/test_main.py

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env python
+
+import unittest
+import numpy as np
+from mecode import G
+
+
+g = G(
+    print_lines=False,
+    aerotech_include=False,
+    direct_write=True,
+    direct_write_mode='serial',
+    #printer_port="/dev/tty.usbmodem1411",
+    printer_port="COM6",
+)
+
+
+def read_profilometer(samples=1):
+    samples = np.floor(np.log2(samples))
```

Comment:
I believe (correct me if I'm wrong) this is just to prevent an extremely large value of samples being used. The floor function just makes sure there's no decimal value by choosing the largest previous value. The floor of 2.5 is 2 and the floor value of -2.5 is -3.